### PR TITLE
[Filter/TF-lite] nnapi option @open sesame 10/06 20:45

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add option structure to open tf-lite model, set NNAPI false as default value.

Using NNAPI in tf-lite may make lower performance and unexpected result for some model.
Let the developer select NNAPI option when constructing the pipeline.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc [hunk 1]
```diff
@@ -47,6 +47,19 @@
 #define DBG FALSE
 #endif
 
+/**
+ * @brief Option to open tf-lite model.
+ */
+typedef struct
+{
+  const gchar *model_file; /**< path to tensorflow-lite mofel file */
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc]: updated

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc [hunk 2]
```diff
@@ -822,6 +834,51 @@ TFLiteCore::cacheInOutTensorPtr ()
...
+    len = g_strv_length (strv);
+
+    for (i = 0; i < len; ++i) {
+      gchar **pair = g_strsplit (strv[i], ":", -1);
+
+      if (g_strv_length (pair) > 1) {
+        g_strstrip (pair[0]);
+        g_strstrip (pair[1]);
+
+        if (g_ascii_strcasecmp (pair[0], "UseNNAPI") == 0) {
```
[the remark this answers is not in the document]
Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc:865]: https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-ascii-strcasecmp

It ignores cases.

Anyway, refer to #2782, we need to manage custom properties soon. Please prepare some descriptions/documentations for custom props somewhere in this file.

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc [hunk 3]
```diff
@@ -822,6 +834,51 @@ TFLiteCore::cacheInOutTensorPtr ()
...
+          if (g_ascii_strcasecmp (pair[1], "true") == 0)
+            option->use_nnapi = TRUE;
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc:867]: It's be easier for users if  (case incensitive)
"t*", "1", "e*", "on","y*" are all interpreted as true
(true/1/enabled/on/yes)

**Review by myungjoo — APPROVED** [review]

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc (hunk 3 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc:872]: I will consider common function (move _parse_bool_string() in conf.c)

**Review by anyj0527 — APPROVED** [review]
LGTM 👍🏼👍🏼👍🏼

**Pull request merged**